cmd/snap/cmd_info.go

```diff
@@ -401,9 +401,11 @@ func (x *infoCmd) Execute([]string) error {
 
 		fmt.Fprintf(w, "name:\t%s\n", both.Name)
 		fmt.Fprintf(w, "summary:\t%s\n", formatSummary(both.Summary))
-		// TODO: have publisher; use publisher here,
-		// and additionally print developer if publisher != developer
-		fmt.Fprintf(w, "publisher:\t%s\n", both.Developer)
+		publisher := "-"
+		if both.Publisher != nil {
+			publisher = both.Publisher.Username
```

Comment:
I think `Publisher` should have something like a `ForTerminal() string` that returned a string that best described the publisher for printing to the terminal. That way when we get verified publishers, or badges, or all the other features that are on the whiteboard for publishers, we just change that and get the changes in info and list.

Reply:
it seems a bit premature and more for a follow up though,  from designs we need long and short versions, and we need to take into account if the terminal is a tty or not